addons/mail/static/src/core/common/record.js

```diff
@@ -1207,13 +1243,14 @@ export class Record {
     static _onChange(record, key, callback) {
         let proxy;
         function _observe() {
-            void proxy[key];
-            if (proxy[key] instanceof Object) {
-                void Object.keys(proxy[key]);
+            // access proxy[key] only once to avoid triggering reactive get() many times
+            const val = proxy[key];
+            if (val instanceof Object) {
+                void Object.keys(val);
             }
-            if (proxy[key] instanceof Array) {
-                void proxy[key].length;
-                void proxy[key].forEach((i) => i);
+            if (val instanceof Array) {
```

Comment:
Same as before: since `instanceof` relies on the prototype, for this to work you have to be 10000% sure that all tested arrays have the exact same prototype as `window.Array` in their prototype chain. This is usually not true in a browser environment, where multiple global objects coexist: the global object can be `window`, but also the other `window` from an iframe, or the global of a worker... That's the reason why [`Array.isArray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) was introduced. I'd recommend using it instead.
```suggestion
            if (Array.isArray(val)) {
```